samples/helios/vmware_workflow_using_v2.py

```diff
@@ -0,0 +1,142 @@
+    CreateOrUpdateProtectionGroupRequest,
+)
+from cohesity_sdk.helios_mcm.v2.model.create_protection_group_run_request import (
+    CreateProtectionGroupRunRequest,
+)
+from cohesity_sdk.helios_mcm.v2.model.create_or_update_protection_group_request import (
+    CreateOrUpdateProtectionGroupRequest,
+)
+from cohesity_sdk.helios_mcm.v2.model.vmware_protection_group_object_params import (
+    VmwareProtectionGroupObjectParams,
+)
+from cohesity_sdk.helios_mcm.v2.model.vmware_protection_group_params import (
+    VmwareProtectionGroupParams,
+)
+
+from config import vcenter_name, vcenter_username, vcenter_password
+from library import register_vmware_source
+from helios_client import helios_v2_connector_with_cluster_id
+
+# Initialise a client.
+client = helios_v2_connector_with_cluster_id()
+job_name = "VmwareJob_" + str(random.randint(1, 100))
+
+
+def get_storage_domain_id():
+    domains = client.storage_domain.get_storage_domains().storage_domains or []
+    for domain in domains:
+        if domain.name == "DefaultStorageDomain":
+            return domain.id
+    return -1
+
+
+def get_policy_id():
+    policies = client.policy.get_protection_policies().policies or []
+    for policy in policies:
+        if policy.name == "Bronze":
+            return policy.id
+    return -1
+
+
+def create_job(job_name):
+    try:
+        source_id = None
+        jobs = client.protection_group.get_protection_groups(is_deleted=False)[
+            "protection_groups"
+        ]
+        # Check the job is already available.
+        for job in jobs:
+            if job["name"] == job_name:
+                print("Job %s is already available, skipping creation" % job_name)
+                return job["id"], job
+
+        # Get storage_domain id.
+        print("Fetching storage domains")
+        domain_id = get_storage_domain_id()
+        if domain_id == -1:
+            raise Exception("Storage Domain not available.")
+
+        # Check if the source is already registered.
+        sources = client.source.mcm_get_protection_sources(
+            environments=["kVMware"]
+        ).sources
+        for source in sources:
+            if source.name == vcenter_name:
+                source_id = source.id
+        if not source_id:
+            register_vmware_source(vcenter_name, vcenter_username, vcenter_password)
+        body = CreateOrUpdateProtectionGroupRequest(
+            storage_domain_id=domain_id,
+            policy_id=get_policy_id(),
+            name=job_name,
+            environment="kVMware",
+            vmware_params=VmwareProtectionGroupParams(
+                objects=[VmwareProtectionGroupObjectParams(id=get_objects_by_vmname())]
+            ),
+        )
+        resp = client.protection_group.create_protection_group(body)
+        return resp
+    except Exception as err:
+        print(err)
+        exit()
+
+
+def get_objects_by_vmname():
+    vm_size = 99999999999
```

Comment:
Why is this logic this way ?

Reply:
To find a VM of smaller size

Reply:
Leave a comment with the size of the VM defined, also this could be constant upfront , for users to change the value.

Reply:
okay